### DISCUSSION TITLE
Orientation dependence of mean GND density

### DISCUSSION THREAD

**Discussion by mfuhr-mpg** [mtex-toolbox/mtex#1339] [Ask Anything] [answered]
Dear all,

I would like to investigate the orientation dependence of the mean GND density per grain for an orientation map acquired using EBSD.
More precisely, I would like to plot the mean GND density calculated for a single grain in an inverse pole figure, i.e. I want to use the meanOrientation of the grain for determining the position of the data point within the ipdf and the mean GND density as the value for creating a contour plot.

Thanks in advance for your answer to that naïve question!

Best regards
Max

**Comment by kilir** [comment, marked as answer]
Hi,
in general, the data, in your case the mean GND goes as second argument `plotIPDF(ori,data,r)`. You can contour those using e.g.  the 'contour' option or `interp` manually.
Hope that helps.
Cheers,
Rüdiger

**Reply by mfuhr-mpg** [reply]
Hi Rüdiger,
that´s exactly what I was looking for!
Thanks once more for the excellent support.
Best
Max

**Reply by matheusGariglio** [reply]
Hi Rüdiger,

When I try to contour the data using 'contourf', the following error message appears :

```
Error using reshape
Number of elements must not change. Use [] as one of the size inputs to automatically calculate the appropriate size for that dimension.

Error in S2FunHarmonic.approximation (line 38)
y = reshape(yy, [length(nodes) s(2:end)]);

Error in vector3d/interp (line 55)
    sF = S2FunHarmonic.approximation(v, y, varargin{:});

Error in vector3d/smooth (line 51)
      cdata = interp(v,cdata,S2G,'cutOutside',varargin{:});

Error in Miller/smooth (line 25)
[varargout{1:nargout}] = smooth@vector3d(m,varargin{:},sR,m.CS);

Error in vector3d/contourf (line 31)
  [varargout{1:nargout}] = v.smooth(data,'contours',10,'LineStyle','-','LineColor','k',varargin{:});

Error in vector3d/plot (line 111)
    [varargout{1:nargout}] = v.contourf(varargin{:});

Error in vector3d/subsref (line 20)
    [varargout{1:nargout}] = builtin('subsref',v,s);

Error in orientation/plotIPDF (line 82)
  [~,cax] = h.plot(repmat(data,1,length(rSym)),'symmetrised',...
```

Do you know why?

Cheers,

Matheus
